packages/app-store/_utils/createAppCredential.ts

```diff
@@ -0,0 +1,57 @@
+import type { NextApiRequest } from "next";
+
+import prisma from "@calcom/prisma";
+
+import { decodeOAuthState } from "./decodeOAuthState";
+
+/**
+ * This function is used to create app credentials for either a user or a team
+ *
+ * @param appData information about the app
+ * @param appData.type the app slug
+ * @param appData.appId the app slug
+ * @param key the keys for the app's credentials
+ * @param req the request object from the API call. Used to determine if the credential belongs to a user or a team
+ */
+const createAppCredential = async (
+  appData: { type: string; appId: string },
+  key: unknown,
+  req: NextApiRequest
+) => {
+  const userId = req.session?.user.id;
+  // For OAuth flows, see if a teamId was passed through the state
+  const state = decodeOAuthState(req);
+  if (state?.teamId) {
+    // Check that the user belongs to the team
+    const team = await prisma.team.findUnique({
+      where: { id: state.teamId },
+      select: { id: true, members: { select: { userId: true } } },
+    });
```

Comment:
Havent tested but this should allow us to leave the search in the prisma query instead of in the IF below (L:31) 

```suggestion
    const team = await prisma.team.findUnique({
     where:{
      id:state.teamId,
      members:{
        some:{
          userId:ctx.user.id
        }
      }
    },
     select: { id: true, members: { select: { userId: true } } },
    });
```

Reply:
Good catch, made the change.

Reply:
@hariombalhara change made in `createOAuthAppCredentials`